tests/unit/test_charm.py

```diff
@@ -730,3 +749,219 @@ def test_when_remove_then_external_service_is_deleted(self, patch_client):
+    @patch("ops.model.Container.push", new=Mock)
+    @patch("ops.model.Container.exec", new=MagicMock)
+    @patch("ops.model.Container.pull", new=Mock)
+    @patch("ops.model.Container.exists")
+    def test_given_container_can_connect_bessd_pebble_ready_when_core_net_mtu_config_changed_to_different_valid_values_then_delete_pod_called_twice(  # noqa: E501
+        self,
+        patch_exists,
+        patch_delete_pod,
+        patch_multus_is_ready,
+        patch_list_na_definitions,
+        patch_nad_created_by_charm,
+    ):
+        patch_exists.return_value = True
+        patch_multus_is_ready.return_value = True
+        self.harness.set_can_connect(container="bessd", val=True)
+        self.harness.set_can_connect(container="pfcp-agent", val=True)
+        self.harness.set_leader(is_leader=True)
+        patch_nad_created_by_charm.return_value = True
+        original_nads = self.harness.charm._network_attachment_definitions_from_config()
+        patch_list_na_definitions.return_value = original_nads
+        self.harness.update_config(key_values={"core-interface-mtu-size": VALID_MTU_SIZE_1})
+        modified_nads = self.harness.charm._network_attachment_definitions_from_config()
+        patch_list_na_definitions.return_value = modified_nads
+        self.harness.update_config(key_values={"core-interface-mtu-size": VALID_MTU_SIZE_2})
+        self.assertEqual(patch_delete_pod.call_count, 2)
+
+    @patch(f"{MULTUS_LIBRARY_PATH}.KubernetesClient", new=Mock)
+    @patch(
+        f"{MULTUS_LIBRARY_PATH}.KubernetesMultusCharmLib._network_attachment_definition_created_by_charm"  # noqa: E501, W505
+    )
+    @patch(f"{MULTUS_LIBRARY_PATH}.KubernetesClient.list_network_attachment_definitions")
+    @patch(f"{MULTUS_LIBRARY_PATH}.KubernetesMultusCharmLib.is_ready")
+    @patch(f"{MULTUS_LIBRARY_PATH}.KubernetesClient.delete_pod")
+    @patch("ops.model.Container.push", new=Mock)
+    @patch("ops.model.Container.exec", new=MagicMock)
+    @patch("ops.model.Container.pull", new=Mock)
+    @patch("ops.model.Container.exists")
+    def test_given_container_can_connect_bessd_pebble_ready_when_core_net_mtu_config_changed_to_same_valid_value_multiple_times_then_delete_pod_called_once(  # noqa: E501
+        self,
+        patch_exists,
+        patch_delete_pod,
+        patch_multus_is_ready,
+        patch_list_na_definitions,
+        patch_nad_created_by_charm,
+    ):
+        """Delete pod is called for the first config change, setting the same config value does not trigger pod restarts."""  # noqa: E501, W505
+        patch_exists.return_value = True
+        patch_multus_is_ready.return_value = True
+        self.harness.set_can_connect(container="bessd", val=True)
+        self.harness.set_can_connect(container="pfcp-agent", val=True)
+        self.harness.set_leader(is_leader=True)
+        patch_nad_created_by_charm.return_value = True
+        original_nads = self.harness.charm._network_attachment_definitions_from_config()
```

Comment:
We should not patch our private methods.

In the tests we should have no reference to private methods except the top level event handlers (ex. `_on_install`). This applies at a couple of places.

Reply:
Done